webdriver/client.py

```diff
@@ -385,12 +377,13 @@ def send_command(self, method, uri, body=None):
 
         url = urlparse.urljoin("session/%s/" % self.session_id, uri)
         response = self.send_raw_command(method, url, body)
+        value = response.body["value"]
 
-        rv = response.body["value"]
-        if not rv:
-            rv = None
+        if response.status != 200:
```

Comment:
I think as a rule, any user of the client API should be right to expect the use of a method to raise if the remote end returns an error. In other words, if the user calls `Session.status` or `Session.start` and the driver for inexplicable reasons returns an error, this must throw.

Maybe the solution is to make each consumer of `send_command` produce the full URI for the command. This is what we already do for [`client.Element`](https://github.com/w3c/wdclient/blob/master/webdriver/client.py#L547), which requires an additional `/element/{element id}` part:

```py
def url(self, suffix):
    return "element/%s/%s" % (self.id, suffix)

…

@command
def click(self):
    self.session.send_command("POST", self.url("click"), {})
```

Admittedly `self.url` isn’t such a great name, but applying the same idea to the `Session` class:

```py
@command
def forward(self):
    return self.send_command("POST", self._session_uri("forward"))
```

Yet another possible approach is to predefine the the endpoints as URI templates. This is common in Selenium client bindings because it allows decoupling of the API from the endpoints so that they can support multiple backend protocols (Selenium vs. W3C WebDriver wire protocols):

```rb
COMMANDS = {
    new_session: [:post, 'session'.freeze],
    get_current_url: [:get, 'session/:session_id/url'.freeze],
    get: [:post, 'session/:session_id/url'.freeze],
   …
}
```